serveur-web/api-v2/controller/sensor.py

```diff
@@ -0,0 +1,182 @@
+	Renvoie une liste de senseurs avec leurs noms
+	"""
+	def get_sensors():
+		db = Database.get_instance()
+		# Sélectionner les dernières données des senseurs
+		temperature = db.execute("SELECT * FROM valeurs WHERE senseur='Temperature' ORDER BY date DESC LIMIT 1;")
+		humidite = db.execute("SELECT * FROM valeurs WHERE senseur='Humidite' ORDER BY date DESC LIMIT 1;")
+
+		senseurs = [
+			{
+				"id" : 1,
+				"date" : temperature[0][0],
+				"name" : temperature[0][1],
+				"value" : temperature[0][2]
+			},
+			{
+				"id" : 2,
+				"date" : humidite[0][0],
+				"name" :  humidite[0][1],
+				"value" : humidite[0][2]
+			}
+		]
+
+		return {"sensors": senseurs }
+
+
+	"""
+	UPDATE LIGHTS
+	Modifier la luminosité de chaque couleur de DEL
+	"""
+	def update_lights(request):
+		red = request.data['red']
+		blue = request.data['blue']
+		white = request.data['white']
+
+		# Mettre à jour leur valeur dans la base de données
+		db = Database.get_instance()
+		datenow = str(datetime.now(timezone.utc))
+		db.execute("INSERT INTO valeurs(date, senseur, valeur) VALUES ('" + datenow + "', 'Red', '" + str(red) + "');")
+		db.execute("INSERT INTO valeurs(date, senseur, valeur) VALUES ('" + datenow + "', 'Blue', '" + str(blue) + "');")
+		db.execute("INSERT INTO valeurs(date, senseur, valeur) VALUES ('" + datenow + "', 'White', '" + str(white) + "');")
+
+		# Command
+		command = SensorController.createCommandLight(white, blue, red)
+		db.execute("INSERT INTO filecommandes(date, command) VALUES ('" + datenow + "', '" + command + "');")
+		return ('', 204)
+
+
+	"""
+	GET LIGHTS
+	Renvoie les valeurs de luminosité de chaque couleur de DEL
+	"""
+	def get_lights():
+		db = Database.get_instance()
+		# Sélectionner les dernières données de luminosité
+		red = db.execute("SELECT * FROM valeurs WHERE senseur='Red' ORDER BY date DESC LIMIT 1;")
+		blue = db.execute("SELECT * FROM valeurs WHERE senseur='Blue' ORDER BY date DESC LIMIT 1;")
+		white = db.execute("SELECT * FROM valeurs WHERE senseur='White' ORDER BY date DESC LIMIT 1;")
+
+		lights = [
+			{
+				"id" : 1,
+				"date" : red[0][0],
+				"name" : red[0][1],
+				"value" : red[0][2]
+			},
+			{
+				"id" : 2,
+				"date" : blue[0][0],
+				"name" :  blue[0][1],
+				"value" : blue[0][2]
+			},
+			{
+				"id" : 3,
+				"date" : white[0][0],
+				"name" :  white[0][1],
+				"value" : white[0][2]
+			}
+		]
+
+		return {"lights": lights}
+
+
+	"""
+	UPDATE FANS
+	Modifier la vitesse d'un ventilateur
+	"""
+	def update_fans(request):
+		# S'il manque des virgules (,) au JSON, il ce peut que le code s'arrête ici.
+		reqdata = request.data
+
+
+		fanl = reqdata.get('fanl')
+		fanh = reqdata.get('fanh')
+
+		datenow = str(datetime.now(timezone.utc))
+
+		# Ventilateur latéral
+		if fanl is not None:
+			# Mettre à jour leur valeur dans la base de données
+			db = Database.get_instance()
+			db.execute("INSERT INTO valeurs(date, senseur, valeur) VALUES ('" + datenow + "', 'FanL', '" + str(fanl) + "');")
```

Comment:
Risque d'injection SQL ici aussi

Reply:
y'a de bonne suggestion ici : https://blog.sqreen.io/preventing-sql-injections-in-python/

Reply:
ça serait excellent je pense!